proto/store/setting.proto

```diff
@@ -205,11 +210,68 @@ message SemanticTypesSetting {
     string title = 2;
     // the description of the semantic type, it can be empty.
     string description = 3;
-    // the partial mask algorithm id for the semantic type, if it is empty, should use the default partial mask algorithm.
+    // the partial mask algorithm id for the semantic type, if it is empty,
+    // should use the default partial mask algorithm.
     string partial_mask_algorithm_id = 4;
-    // the full mask algorithm id for the semantic type, if it is empty, should use the default full mask algorithm.
+    // the full mask algorithm id for the semantic type, if it is empty, should
+    // use the default full mask algorithm.
     string full_mask_algorithm_id = 5;
   }
 
   repeated SemanticType types = 1;
 }
+
+message MaskingAlgorithmSetting {
+  message Algorithm {
+    // id is the uuid for masking algorithm.
+    string id = 1;
+    // title is the title for masking algorithm.
+    string title = 2;
+    // description is the description for masking algorithm.
+    string description = 3;
+
+    enum Category {
+      CATEGORY_UNSPECIFIED = 0;
+      MASKING = 1;
+      HASHING = 2;
+    }
+    // category is the category for masking algorithm, the range of accepted Payload is decided by the category.
+    // MASKING: FullMaskingPayload, RangePartialMaskingPayload
+    // HASHING: MD5Payload
+    Category category = 4;
+
+    message FullMaskingPayload {
+      // replace_with is the string used to replace the original value, the
+      // max length of the string is 16 bytes.
+      string replace_with = 1;
+    }
+
+    message RangePartialMaskingPayload {
```

Comment:
I'm trying to avoid the word "Payload" in the message name because it's not meaningful. Also it's not useful to put word "Partial" because it's a choice of masking algorithm.

message RangeMask {}
message FullMask {}
message MD5Mask {}

If we really want to indicate it's a payload, use the word "config", e.g. RangeMaskConfig.